docs/source/tools/watcher.rst

```diff
@@ -152,3 +152,34 @@ The resulting message will be printed on the command line (and watcher log):
 
     INFO     | Paused cores: (x=1,y=1):brisc
     Press ENTER to unpause core(s) and continue...
+
+Ring Buffer
+-----------
+A small ring buffer is available on each core, accessible via the `RING_BUFFER_PUSH()` macro.
+This ring buffer has 31 `uint32_t` elements, and when more than the max amounts of elements
+are pushed into the buffer, the oldest are overwritten. When the watcher is disabled, the ring
+buffer is still present, but any writes to it are compiled out. An example of pushing data to the
+rung buffer, and the resulting log is shown below.
+
+Important: the ring buffer does not have any synchronization for writes between difference RISCs on
+the same core. Calling `RING_BUFFER_PUSH()` from different RISCs in the same core at the same time
+mat result in undefined behaviour.
+
+.. code-block::
+
+    #include "debug/ring_buffer.h"
+
+    void kernel_main() {
+        for (uint32_t idx = 0; idx < 40; idx++) {
+            RING_BUFFER_PUSH(idx+1);
```

Comment:
let's qualify the name w/ either WATCHER or DEBUG or some such.  Too many ring buffers...